apps/prairielearn/src/cron/sendExternalGraderDeadLetters.js

```diff
@@ -1,49 +1,43 @@
+// @ts-check
 const ERR = require('async-stacktrace');
 const async = require('async');
-const AWS = require('aws-sdk');
+const {
+  SQSClient,
+  GetQueueUrlCommand,
+  ReceiveMessageCommand,
+  DeleteMessageCommand,
+} = require('@aws-sdk/client-sqs');
+const { logger } = require('@prairielearn/logger');
 
 const { config } = require('../lib/config');
-const { logger } = require('@prairielearn/logger');
 const opsbot = require('../lib/opsbot');
 
 // After loading the queue url for the first time, we'll cache it here
 const QUEUE_URLS = {};
 
-module.exports = {};
-
 module.exports.run = (callback) => {
   if (!opsbot.canSendMessages()) return callback(null);
-  if (!config.externalGradingUseAws) return callback(null);
 
-  const sqs = new AWS.SQS(config.awsServiceGlobalOptions);
+  const jobsDeadLetterQueueName = config.externalGradingJobsDeadLetterQueueName;
+  const resultsDeadLetterQueueName = config.externalGradingResultsDeadLetterQueueName;
+  if (!config.externalGradingUseAws || !jobsDeadLetterQueueName || !resultsDeadLetterQueueName) {
+    return callback(null);
+  }
+
+  const sqs = new SQSClient({
+    region: config.awsRegion,
+    ...config.awsServiceGlobalOptions,
```

Comment:
AFAICT this was added for [localstack](https://localstack.cloud/) compatibility? Do any of us use that?

Reply:
Not any more. At some point we were trying it for local S3 mocking, I think?

Reply:
Yeah, I guess [s3rver](https://github.com/jamhall/s3rver) is what we ultimately landed on?